lib/dangermattic/plugins/podfile_checks.rb

```diff
@@ -32,37 +32,85 @@ class PodfileChecks < Plugin
     #
     # @return [void]
     def check_podfile_does_not_have_commit_references(podfile_lock_path: DEFAULT_PODFILE_LOCK_PATH)
+      check_podfile_does_not_match(
+        regexp: COMMIT_REFERENCE_REGEXP,
+        podfile_lock_path: podfile_lock_path,
+        match_found_message_generator: ->(matches) { "Podfile reference(s) to a commit hash:\n```#{matches.join("\n")}```" }
+      )
+    end
+
+    # Check for Podfile references to commit hashes in the Podfile.lock in a pull request.
+    #
+    # @return [void]
+    def check_podfile_diff_does_not_have_commit_references
+      warning_message = 'This PR adds a Podfile reference to a commit hash:'
+      check_podfile_diff_entries_do_not_match(regexp: COMMIT_REFERENCE_REGEXP, match_found_message: warning_message)
+    end
```

Comment:
I basically just abstracted the previous implementation to inject the message and the RegExp then used different pairs for commit and branch checks.

Reply:
Great idea and resulting implementation 👍